src/components/Calculator/logic/calculate.test.js

```diff
@@ -0,0 +1,19 @@
+const calculate = require('./calculate').default;
+
+test('AC button should reset the calculator state', () => {
+  const initialState = {
+    total: '100',
+    text: '500',
+    operation: '+',
+  };
+
+  const expectedState = {
+    total: null,
+    text: null,
+    operation: null,
+  };
+
+  const newState = calculate(initialState, 'AC');
+
+  expect(newState).toEqual(expectedState);
```

Comment:
- [ ] React Testing Library should be used to simulate user interaction for this test